src/core.py

```diff
@@ -60,6 +61,30 @@ def process_dir(root_dir, curr_dir, args, template=None):
     if os.path.exists(local_template_path):
         template = Template(local_template_path, PROCESSOR_MANAGER.processors)
 
+    # looking for pdf files in current dir and saving them as into a new folder named the same as pdf file and saving individual pages as images
+    pdf_files = [f for f in curr_dir.glob("*.pdf")]
+
+    if pdf_files:
+        for pdf in pdf_files:
+            path = str(pdf)
+            obj = pdfium.PdfDocument(path)
+            path = path[:-4]
+            try:
+                os.mkdir(path)
+            except FileExistsError:
+                print(f"Warning, folder with name {path} already exists")
+
+            for page_num in range(len(obj)):
+                page = obj.get_page(page_num)
+                pil_image = page.render_topil(
+                    scale=1,
+                    rotation=0,
+                    crop=(0, 0, 0, 0),
+                    greyscale=False,
+                    optimise_mode=pdfium.OptimiseMode.NONE,
+                )
+                pil_image.save(f"{path}/{page_num}.png")
```

Comment:
we'd want to avoid saving temporary image. This increases file IO and slows down the speed. Can you figure out a way to do this "in-memory" using a variable only?